codeowners/__init__.py

```diff
@@ -76,7 +76,10 @@ def path_to_regex(pattern: str) -> Pattern[str]:
                 if (left_anchored or leading_slash) and (
                     right_anchored or trailing_slash
                 ):
-                    regex += ".*"
+                    if trailing_slash:
+                        regex += "(?:.*/)?"
```

Comment:
Can you add a test to cover this?

Reply:
Done. I tried to hit all existing test cases where there was a `**` but expanding to `.*` rather than `(**/)?` would have been incorrect. I didn't change any of the form `**/*blah` since they should behave the same before and after this patch, although for different reasons.